### PR TITLE
♻️ Condense polyfills and lock down imports

### PR DESCRIPTION
**Pull request by rcebulko**
Moves all polyfills into `src/polyfills` and adds a lint rule preventing imports outside of polyfills/core. In preparation for enabling/passing typechecking on `src/polyfills`

### CODE REVIEW

File: src/runtime.js [hunk 1]
```diff
@@ -262,8 +262,13 @@ function adoptShared(global, callback) {
   }
 
   // Some deferred polyfills.
-  scheduleInObUpgradeIfNeeded(global);
-  scheduleResObUpgradeIfNeeded(global);
+  const extensionsFor = Services.extensionsFor(global);
```
Comment (samouri) [inline src/runtime.js:265]: Clever way to avoid Services for now ⭐

**Review by samouri — APPROVED** [review]

File: src/runtime.js (hunk 1 above)
Comment (rcebulko) [reply src/runtime.js:265]: Yeah! I figure this barely is owned by polyfills themselves, and this is literally the only place those methods were used, so it seemed like an efficient way to untangle some AMP dependencies

**Review by rsimha — APPROVED** [review]
Build changes 👍

File: src/polyfills/index.js [hunk 2]
```diff
@@ -14,25 +14,25 @@
  * limitations under the License.
  */
 
-/** @fileoverview */
+/** @fileoverview Installs polyfills depending on build environment. */
```
Comment (rsimha) [inline src/polyfills/index.js:17]: Looks like there are [~300 comments](https://github.com/ampproject/amphtml/search?utf8=%E2%9C%93&q=%22src%2Fpolyfills.js%22) that read `// src/polyfills.js must be the first import.`, that could do with an update. They mostly live in 3p code.

Comment (rcebulko) [reply src/polyfills/index.js:17]: Yep I've seen em all. I plan to update all of those in a separate PR so Justin or Kris can rubber-stamp without needing to look

**Review by caroqliu — APPROVED** [review]

**Pull request merged**